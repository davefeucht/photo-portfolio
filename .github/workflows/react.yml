# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions

name: React CI/CD

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:
  build:

    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [18.x]

    steps:
    - name: Checkout code
      uses: actions/checkout@v2

    - name: Install Node.js ${{ matrix.node-version }}
      uses: actions/setup-node@v1
      with:
        node-version: ${{ matrix.node-version }}

    - name: Enable corepack
      run: corepack enable

    - name: Install dependencies
      run: yarn install

    - name: Build project
      run: npm run build --if-present

    - name: Run tests
      run: npm test
