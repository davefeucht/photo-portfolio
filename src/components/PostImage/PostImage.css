.post-image {
  position: relative;
  grid-area: content;
  display: grid;
  grid-template-columns: 100%;
  grid-template-rows: 100%;
  transition: width var(--modal-transition-time), height var(--modal-transition-time)
}

.post-image img {
  transition: all var(--modal-transition-time), height var(--modal-transition-time)
}