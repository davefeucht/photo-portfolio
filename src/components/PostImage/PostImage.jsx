/****************
* PostImage component displays the image for a Post
****************/

import React, { useEffect } from "react";
import { observer } from 'mobx-react';
import './PostImage.css';

const PostImage = observer(({ stateStore }) => {
  return(
    <div className="post-image">
      <img src={stateStore.visiblePost.fullImageUrl}></img>
    </div>
  )
});

PostImage.displayName = 'PostImage';

export default PostImage;