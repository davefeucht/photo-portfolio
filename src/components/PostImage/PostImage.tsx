/** **************
* PostImage component displays the image for a Post
*************** */

import './PostImage.css';

import { observer } from 'mobx-react';
import * as React from 'react';
import { useState } from 'react';

import PostNavigationArrow from '../PostNavigationArrow/PostNavigationArrow';

interface PostImageProps {
    imageUrl: string,
    postHeight: number,
    previousPost: string,
    nextPost: string
}

const PostImage: React.FC<PostImageProps> = ({
    imageUrl,
    postHeight,
    previousPost,
    nextPost
}) => {
    const [arrowsVisible, setArrowsVisible] = useState<boolean>(false);

    const onMouseOverHandler = () => {
        setArrowsVisible(true);
    };

    const onMouseOutHandler = () => {
        setArrowsVisible(false);
    };

    return (
        <div className="post-image" onMouseOver={onMouseOverHandler.bind(this)} onMouseOut={onMouseOutHandler.bind(this)}>
            {arrowsVisible && <PostNavigationArrow postHeight={postHeight} direction="previous" postId={previousPost} />}
            <img src={imageUrl} />
            {arrowsVisible && <PostNavigationArrow postHeight={postHeight} direction="next" postId={nextPost} />}
        </div>
    );
};

PostImage.displayName = 'PostImage';

export default observer(PostImage);
