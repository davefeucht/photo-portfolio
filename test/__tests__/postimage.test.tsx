import { render } from '@testing-library/react';
import PostImage from 'components/PostImage/PostImage';
import * as React from 'react';
import { useRef } from 'react';
import { MemoryRouter } from 'react-router-dom';

const previousPostId = 34;
const nextPostId = 36;
const imageUrl = 'https://throughapinhole.com/wp-content/uploads/2018/07/DSC_1508.jpg';

const WrapperComponent: React.FC = () => {
    const ref = useRef<HTMLImageElement>(null);
    return (
        <MemoryRouter>
            <PostImage imageUrl={imageUrl} imageHeight={150} previousPost={previousPostId} nextPost={nextPostId} ref={ref} />
        </MemoryRouter>
    );
};

test('PostImage displays', async () => {
    const { container } = render(
        <WrapperComponent />
    );
    expect(container.firstChild).toMatchSnapshot();
});
