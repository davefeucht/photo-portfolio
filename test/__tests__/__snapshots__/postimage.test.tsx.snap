// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostImage PostImage displays 1`] = `
<div
  aria-label="post-image"
  class="post-image"
>
  <img
    src="https://throughapinhole.com/wp-content/uploads/2018/07/DSC_1508.jpg"
  />
</div>
`;
